api/turing/cluster/controller.go

```diff
@@ -165,29 +162,22 @@ func newController(clusterCfg clusterConfig) (Controller, error) {
 // key is the env name and the value is the corresponding controller.
 func InitClusterControllers(
 	cfg *config.Config,
-	environmentClusterMap map[string]string,
-	vaultClient vault.Client,
+	environmentClusterMap map[string]*mlpcluster.K8sConfig,
 ) (map[string]Controller, error) {
 	// For each supported environment, init controller
 	controllers := make(map[string]Controller)
-	for envName, clusterName := range environmentClusterMap {
+	for envName, k := range environmentClusterMap {
 		clusterCfg := clusterConfig{
-			ClusterName: clusterName,
+			ClusterName: k.Name,
 		}
 
 		if cfg.ClusterConfig.InClusterConfig {
 			clusterCfg.InClusterConfig = true
 		} else {
-			clusterSecret, err := vaultClient.GetClusterSecret(clusterName)
-			if err != nil {
-				return nil, errors.Wrapf(err,
-					"unable to get cluster secret for cluster: %s", clusterName)
-			}
+			creds := mlpcluster.NewK8sClusterCreds(k)
 
-			clusterCfg.Host = clusterSecret.Endpoint
-			clusterCfg.CACert = clusterSecret.CaCert
-			clusterCfg.ClientCert = clusterSecret.ClientCert
-			clusterCfg.ClientKey = clusterSecret.ClientKey
+			clusterCfg.Host = k.Cluster.Server
```

Comment:
The `Host` can be removed here. It is won't be used in `newController()` anymore

Reply:
Actually, `clusterConfig` structure will lose most of its fields